pkg/network/master/vnids.go

```diff
@@ -39,8 +41,11 @@ func newMasterVNIDMap(allowRenumbering bool) *masterVNIDMap {
 	}
 
 	return &masterVNIDMap{
-		netIDManager:     pnetid.NewInMemory(netIDRange),
-		adminNamespaces:  sets.NewString(metav1.NamespaceDefault),
+		netIDManager: pnetid.NewInMemory(netIDRange),
+		// the HostNetworkNamespace should also be assigned vnid 0 so that it can
+		// be treated the same as default namespace to allow specifying
+		// network policies for host-network traffic.
+		adminNamespaces:  sets.NewString(metav1.NamespaceDefault, HostNetworkNamespace),
```

Comment:
This won't work. As mentioned in the enhancement, the NetworkPolicy code does not allow two namespaces to have the same VNID, and so `openshift-host-network` can't _actually_ have VNID 0. (eg, `networkPolicyPlugin.namespaces` is a map from VNID to namespace; if you have two namespaces with the same VNID, then only one of them would get added to `namespaces` and it's not even well-defined which one).

I _think_ what you want to do is hack `selectNamespacesInternal` in `networkpolicy.go` so that if `HostNetworkNamespace` matches, it sets `match.matches[npns.name] = 0` instead of `match.matches[npns.name] = vnid`. I _think_ that should have the right effect, and not have any side effects...

Reply:
@danwinship updated the PR with your suggested change. PTAL.